src/backend/distributed/utils/colocation_utils.c

```diff
@@ -988,7 +988,6 @@ DeleteColocationGroup(uint32 colocationId)
 	{
 		simple_heap_delete(pgDistColocation, &(heapTuple->t_self));
 
-		CatalogUpdateIndexes(pgDistColocation, heapTuple);
```

Comment:
Actually, the bug is fixed with this change. The others could also lead to some other issues, so included in the PR